### PR TITLE
fix filename length restriction

### PR DESCRIPTION
**Pull request by GWarsow**
this is a draft for fixing the filename length restriction (512 chars) for files generated by `pdf()` in `chrom_diff.r`

### CODE REVIEW

File: resources/analysisTools/qcPipeline/workflowLib.sh [hunk 1]
```diff
@@ -57,7 +57,11 @@ getBigScratchDirectory () {
     echo "$scratchDir"
 }
 
-
+getFilePathOnRoddyScratch () {
+    local sourceLocation="${1}"
+    basename=$(basename ${sourceLocation})
+    echo "${RODDY_SCRATCH}/${basename}"
```
Comment (vinjana) [inline resources/analysisTools/qcPipeline/workflowLib.sh:63]: 1. No argument should result in error.
2. `basename` must be local or will be global.
3. Don't use `${...}` unless necessary (e.g. if default value, error messages, or any other Bash feature that requires that syntax. The `{}` clutter the picture)
```suggestion
    local sourceLocation="${1:?No source location}"
    local basename="$(basename "$sourceLocation")"
    echo "$RODDY_SCRATCH/$basename"
```

File: resources/analysisTools/qcPipeline/bwaMemSortSlim.sh [hunk 2]
```diff
@@ -273,4 +273,5 @@ mv ${FILENAME_QCJSON}.tmp ${FILENAME_QCJSON} || throw 27 "Could not move file"
 
 ${RSCRIPT_BINARY} ${TOOL_INSERT_SIZE_PLOT_SCRIPT} ${FILENAME_ISIZES_MATRIX} ${FILENAME_ISIZES_STATISTICS} ${FILENAME_ISIZES_PLOT}_temp "PE insertsize of ${bamname}" && mv ${FILENAME_ISIZES_PLOT}_temp ${FILENAME_ISIZES_PLOT} || ( echo "Error from insert sizes plotter" && exit 22)
 
-${RSCRIPT_BINARY} ${TOOL_PLOT_DIFFCHROM} -i "$FILENAME_DIFFCHROM_MATRIX" -s "$FILENAME_DIFFCHROM_STATISTICS" -o "${FILENAME_DIFFCHROM_PLOT}_temp" && mv  ${FILENAME_DIFFCHROM_PLOT}_temp ${FILENAME_DIFFCHROM_PLOT} || ( echo "Error from chrom_diff.r" && exit 23)
+FILENAME_DIFFCHROM_PLOT_TMP=$(getFilePathOnRoddyScratch "${FILENAME_DIFFCHROM_PLOT}")
+${RSCRIPT_BINARY} ${TOOL_PLOT_DIFFCHROM} -i "$FILENAME_DIFFCHROM_MATRIX" -s "$FILENAME_DIFFCHROM_STATISTICS" -o "${FILENAME_DIFFCHROM_PLOT_TMP}" && mv ${FILENAME_DIFFCHROM_PLOT_TMP} ${FILENAME_DIFFCHROM_PLOT} || ( echo "Error from chrom_diff.r" && exit 23)
```
Comment (vinjana) [inline resources/analysisTools/qcPipeline/bwaMemSortSlim.sh:277]: 1. Quote subshell returns used as variable value
2. Split superlong line into multiline command.

```suggestion
FILENAME_DIFFCHROM_PLOT_TMP="$(getFilePathOnRoddyScratch "$FILENAME_DIFFCHROM_PLOT")"
$RSCRIPT_BINARY "$TOOL_PLOT_DIFFCHROM" \
  -i "$FILENAME_DIFFCHROM_MATRIX" \
  -s "$FILENAME_DIFFCHROM_STATISTICS" \
  -o "$FILENAME_DIFFCHROM_PLOT_TMP" \
  && mv "$FILENAME_DIFFCHROM_PLOT_TMP" "$FILENAME_DIFFCHROM_PLOT" \
  || ( echo "Error from chrom_diff.r" && exit 23)
```

**Review by vinjana — COMMENTED** [review]
Content seems OK. Otherwise small things.

**Comment by vinjana** [comment]
@GWarsow We agreed to close this PR, right?

**Comment by GWarsow** [comment]
True, I will close it now.

**Pull request closed without merging**